### PR TITLE
Feat/python: Pylint errors

### LINKED ISSUE #1217: Fix python CI failures

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Some examples need to be changed to the Dacyte `from_dict(a, b)` because pylint not properly supporting decorators yet

fixes #1217 .

### CODE REVIEW

**Review by DaughterOfMars — APPROVED** [review]

File: bindings/python/examples/.pylintrc [hunk 1]
```diff
@@ -5,6 +5,7 @@ disable=missing-module-docstring,
 	invalid-name, # files that starts with number
 	duplicate-code,
 	consider-using-with,
-	consider-using-f-string
+	consider-using-f-string,
+	fixme
```
Comment (thibault-martinez) [inline bindings/python/examples/.pylintrc:9]: Why?

File: bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 # Recovers addresses from example `0_address_generation`.
 addresses_data = json.load(open(ADDRESSES_FILE_PATH, "r", encoding="utf-8"))
-addresses = list(map(lambda x: AccountAddress.from_dict(x), addresses_data))
+addresses = list(map(lambda x: from_dict(data_class=AccountAddress, data=x), addresses_data))
```
Comment (thibault-martinez) [inline bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: Why do we not have the other form for this ?

File: bindings/python/examples/.pylintrc (hunk 1 above)
Comment (kwek20) [reply bindings/python/examples/.pylintrc:9]: it considers TODO an error

File: bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py (hunk 2 above)
Comment (kwek20) [reply bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: see pr description

Comment (thibault-martinez) [reply bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: Yes, doesn't really answer my question. Why are some types allowed to have `Type.from_dict` and some not ?

Comment (kwek20) [reply bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: just the examples that dont allow it

Comment (thibault-martinez) [reply bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: Sorry, can you explain to a complete Python novice why this is the case? Why are examples treated differently ?

Comment (kwek20) [reply bindings/python/examples/wallet/offline_signing/1_prepare_transaction.py:30]: well for starters we only run pylint on the examples 🤷‍♂️ also the decorator of dataclass is a new feature, and not really supported. It is claimed to be better in 3.11, but our ci has 3.10.

Basically this: https://github.com/pylint-dev/pylint/issues/7437

**Review by thibault-martinez — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**